### ISSUE TITLE
4.9.7Adaptation problem

### ISSUE THREAD

**Issue by CmosLee** [fluttercandies/flutter_smart_dialog#188]
### Version Information
- Flutter version: [e.g. v3.19.5]
- flutter_smart_dialog version: [e.g. v4.9.7]

### Describe the bug/requirement
On a Huawei mobile phone, SmartDialog.dismiss() is invalid after SmartDialog.show() is run, and dialog is not closed
The same code and phone will work fine in v4.9.6+2 and earlier versions
### Problem demo
```
 if (Platform.isAndroid) EolPermissionTip.show(tip);
    final PermissionStatus status = await permission.request();
    if (Platform.isAndroid) EolPermissionTip.hide();
```

**Comment by CmosLee** [comment]
The same code and phone will work fine in v4.9.6+2 and earlier versions

**Comment by xdd666t** [comment]
The 4.9.7 version has reengineered the initialization logic, and there may indeed be some issues. Please provide a most simplified demo that can be run directly and reproduce the problem. I need to locate where the problem is.

**Comment by CmosLee** [comment]
Further code after await is invalid, but versions prior to v4.9.6+2 are fine
```
void _showDialog() async {
    SmartDialog.show<void>(
        builder: (_) {
          Widget child = Container(
            height: 80,
            width: 180,
            decoration: BoxDecoration(
              color: Colors.black,
              borderRadius: BorderRadius.circular(10),
            ),
            alignment: Alignment.center,
            child: const Text(
              'easy custom dialog',
              style: TextStyle(color: Colors.white),
            ),
          );
          return SafeArea(child: child);
        },
        keepSingle: true,
        permanent: true);
    await Permission.notification.request();
    _closeDialog();
  }

  void _closeDialog() {
    SmartDialog.dismiss<void>(force: true);
  }
```

**Comment by CmosLee** [comment]
用你代码里的demo就可以复现这个问题，我给你发个附件




放弃天堂ForU
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "fluttercandies/flutter_smart_dialog"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年5月20日(星期一) 下午3:22
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [fluttercandies/flutter_smart_dialog] 4.9.7Adaptation problem (Issue #188)





 
The 4.9.7 version has reengineered the initialization logic, and there may indeed be some issues. Please provide a most simplified demo that can be run directly and reproduce the problem. I need to locate where the problem is.

**Comment by xdd666t** [comment]
- try the new version

```dart
dependencies:
  flutter_smart_dialog: ^4.9.7+1
```

**Comment by CmosLee** [comment]
The problem has been solved

**Comment by CmosLee** [comment]
The issue has been fixed in v4.9.7+1
